packages/dev/core/src/FlowGraph/Blocks/Execution/flowGraphSetPropertyBlock.ts

```diff
@@ -36,35 +27,28 @@ export class FlowGraphSetPropertyBlock<ValueT> extends FlowGraphWithOnDoneExecut
      * Input connection: The value to set on the property.
      */
     public readonly value: FlowGraphDataConnection<ValueT>;
+    /**
+     * Input connection: The template strings to substitute in the path.
+     */
+    public readonly templateStringInputs: FlowGraphDataConnection<number>[] = [];
 
     public constructor(public config: IFlowGraphSetPropertyBlockConfiguration) {
         super(config);
 
         this.value = this._registerDataInput("value", RichTypeAny);
-        this._registerDataInput(config.subString, RichTypeNumber);
-    }
-
-    private _setProperty(target: any, property: string, value: any): void {
-        const splitProp = property.split(".");
-
-        let currentTarget = target;
-        for (let i = 0; i < splitProp.length - 1; i++) {
-            currentTarget = currentTarget[splitProp[i]];
+        for (const templateString of config.path.getTemplateStrings()) {
+            this.templateStringInputs.push(this._registerDataInput(templateString, RichTypeNumber));

```

Comment:
Maybe you could factor this code somewhere as an helper, as you used it several times.

[...] After some thinking, you could create a small class instead of a type for `FlowGraphDataConnection<number>[]` and move the code to fill the array and query the path to this class (pseudo-code, not tested):

```typescript
class XXX // I let you find the right name :)
{
    templateStringInputs: FlowGraphDataConnection<number>[] = [];

    constructor(owner: FlowGraphBlock, protected _path: FlowGraphPath) {
        for (const templateString of this._path.getTemplateStrings()) {
            this.templateStringInputs.push(owner._registerDataInput(templateString, RichTypeNumber));
    }

    setSubstitution(context: FlowGraphContext) {
        for (const templateString of this.templateStringInputs) {
            const value = templateString.getValue(context);
            const name = templateString.name;
            this._path.setTemplateSubstitution(name, value);
        }
    }

    getValue(context: FlowGraphContext) {
        this.setSubstitution(context);
        return this._path.getProperty(context);
    }
}
```
I created a separate `setSubstitution` method because I can see below that you use this loop in the `_execute` method but does not need to evaluate the path. So, you could implement `_execute` like:
```typescript
public _execute(context: FlowGraphContext): void {
    xxx.setSubstitution(context);
    const value = this.value.getValue(context);
    this.config.path.setProperty(context, value);

    this.onDone._activateSignal(context);
}
```

Reply:
@Popov72 Created the class! It's named FlowGraphPathComponent.